conans/model/options.py

```diff
@@ -71,7 +72,7 @@ def __nonzero__(self):
 
     def __getattr__(self, attr):
         if attr not in self._dict:
-            return None
+            raise ConanException(option_not_exist_msg(attr, list(self._dict.keys())))
```

Comment:
Important change behavior, to align with the raise of ``self.option.nonexisting``, this is a bug fix.

Reply:
Yes, recipes with typos will start to break, but I think this has been requested previously to avoid typos.

```
def package_id(self):
    del self.info.options.salkdjfsafldasj
```

maybe @uilianries remembers...

Reply:
OK, lets go for it, and lets see if someone reports their builds breaking because of relying on this bug behavior.